scripts/pre_push_hook_test.py

```diff
@@ -138,29 +154,39 @@ def mock_popen(cmd_tokens, stdout, stderr):  # pylint: disable=unused-argument
         with popen_swap:
             self.assertEqual(pre_push_hook.get_remote_name(), b'upstream')
 
-    def test_get_remote_name_with_error_in_obtaining_remote(self):
-        def mock_communicate():
+    def test_get_remote_name_with_error_in_obtaining_remote(self) -> None:
+        def mock_communicate() -> Tuple[bytes, bytes]:
             return (b'test', b'test_oppia_error')
         process = subprocess.Popen(
             [b'echo', b'test'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        process.communicate = mock_communicate
-        def mock_popen(unused_cmd_tokens, stdout, stderr):  # pylint: disable=unused-argument
+        # Here we use MyPy ignore because here we are assigning a value to the
```

Comment:
This seems odd to me, isn't the return value of subprocess.Popen a file object (not a method)?

Ditto elsewhere.

Reply:
Oh, sorry, I misread -- I thought this comment was saying `process` is a method. 

On rereading I think normally the way we handle this is using self.swap() rather than direct assignment, can we do that here?

Reply:
Done -- now `self.swap()` is used.